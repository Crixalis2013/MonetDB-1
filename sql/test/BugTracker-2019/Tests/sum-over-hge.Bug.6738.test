query IIRR rowsort
select foo,
       sum(foo) over () as s1,
       1.0 * sum(foo) over () as s2,
       1.0 * cast(sum(foo) over () as float) as s3
  from (select 7 as foo union all select 3 as foo) as t
----
3
10
10.000
10.000
7
10
10.000
10.000

query II rowsort
select foo,
       sum(foo) over () / 2
  from (select 7 as foo union all select 3 as foo) as t
----
3
5
7
5

query IIR rowsort
select foo,
       sum(foo) over (order by foo) as sumsum1,
       1.0 * sum(foo) over (order by foo) as cumsum2
  from (values (1),(2),(3),(4),(5),(6)) as t(foo)
----
18 values hashing to cbb9c388514f15352c2308bec1055a02


