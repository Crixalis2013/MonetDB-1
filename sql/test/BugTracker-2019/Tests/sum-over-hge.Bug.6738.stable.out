stdout of test 'sum-over-hge.Bug.6738` in directory 'sql/test/BugTracker-2019` itself:


# 17:13:59 >  
# 17:13:59 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=30631" "--set" "mapi_usock=/var/tmp/mtest-29714/.s.monetdb.30631" "--set" "monet_prompt=" "--forcemito" "--dbpath=/ufs/sjoerd/@Monet-stable/var/MonetDB/mTests_sql_test_BugTracker-2019" "--set" "embedded_c=true"
# 17:13:59 >  

#select foo,
#       sum(foo) over () as s1,
#       1.0 * sum(foo) over () as s2,
#       1.0 * cast(sum(foo) over () as float) as s3
#  from (select 7 as foo union all select 3 as foo) as t;
% .t,	.,	.,	. # table_name
% foo,	s1,	s2,	s3 # name
% tinyint,	hugeint,	decimal,	double # type
% 1,	2,	40,	24 # length
[ 7,	10,	10.00,	10	]
[ 3,	10,	10.00,	10	]
#select foo,
#       sum(foo) over () / 2
#  from (select 7 as foo union all select 3 as foo) as t;
% .t,	. # table_name
% foo,	%10 # name
% tinyint,	hugeint # type
% 1,	1 # length
[ 7,	5	]
[ 3,	5	]
#select foo,
#       sum(foo) over (order by foo) as sumsum1,
#       1.0 * sum(foo) over (order by foo) as cumsum2
#  from (values (1),(2),(3),(4),(5),(6)) as t(foo);
% .t,	.,	. # table_name
% foo,	sumsum1,	cumsum2 # name
% tinyint,	hugeint,	decimal # type
% 1,	2,	40 # length
[ 1,	1,	1.00	]
[ 2,	3,	3.00	]
[ 3,	6,	6.00	]
[ 4,	10,	10.00	]
[ 5,	15,	15.00	]
[ 6,	21,	21.00	]

# 17:14:00 >  
# 17:14:00 >  "Done."
# 17:14:00 >  

